Sandbox/Sandbox.cpp

```diff
@@ -30,67 +40,196 @@ void SandboxLayer::onCreation()
+    {
+        entity->draw();
+    }
+    
+    if (Aquarius::Input::isKeyPressed(Aquarius::Input::KeyCode::Key_m))
+    {
+        m_MenuOpen = true;
+    }
+}
 
-    Aquarius::Renderer::DrawQuad(
-        { 50, 200 },
-        { 50, 50},
-        m_texture.get()
-    );
+void SandboxLayer::onUpdateGUI(Aquarius::timeDelta_t ts)
+{
 
-    Aquarius::Renderer::DrawQuad(
-        { 200, 200 },
-        { 50, 50 },
-        m_texture.get(),
-        &m_uv
-    );
+    if (m_MenuOpen)
+    {
+        ImGui::Begin("Sandbox Demo", &m_MenuOpen);   // Pass a pointer to our bool variable (the window will have a closing button that will clear the bool when clicked)
+        ImGui::Text("Sandbox Demo!");
 
-    Aquarius::Renderer::DrawQuad(
-        { 200, 200 },
-        { 50, 50},
-        m_texture.get()
-    );
+        if (ImGui::CollapsingHeader("About"))
+        {
+            ImGui::Text("This is a sandbox where various features of the engine are tested");
+            ImGui::Text("- WASD Moves the camera around");
+            ImGui::Text("- Arrow up/down to zoom in and out");
+            ImGui::Text("- Initially, there are 10 entities randomly rendered");
+            ImGui::Text("- Entities can be:");
+            ImGui::Text("   - Added one at a time");
+            ImGui::Text("   - Popped one at a time");
+            ImGui::Text("   - Added in bulk (N at a time)");
+            ImGui::Text("   - Cleared (nuked)");
+        }
+
+        if (ImGui::CollapsingHeader("Controls"))
+        {
+            if (ImGui::Button("Generate Single Entity"))
+            {
+                // TODO - Make it so you can configure the entity
+                Entity* ent = new Entity(
+                    { rand() % 800, rand() % 600 },
+                    { 32, 32 },
+                    { (rand() % 255) / 255.0, (rand() % 255) / 255.0, (rand() % 255) / 255.0, 1 },
+                    0
+                );
+                m_entities.push_back(ent);
+            }
+            
+            static int numEntities;
+            ImGui::InputInt("Num Entities", &numEntities);
+            
+            if (ImGui::Button("Generate N Entities"))
+            {
+                for (int i = 0; i < numEntities; i++)
+                {
+                    Entity* ent = new Entity(
+                        { rand() % 800, rand() % 600 },
+                        { 32, 32 },
+                        { (rand() % 255) / 255.0, (rand() % 255) / 255.0, (rand() % 255) / 255.0, 1 },
+                        0
+                    );
+                    m_entities.push_back(ent);
+                }
+            }
+
+            if (ImGui::Button("Pop Entity"))
+            {
+                // TODO - Make it so you can configure the entity
+                if (m_entities.size() != 0)
+                {
+                    Entity* ent = m_entities.back();
+                    delete ent;
+                    ent = nullptr;
+                    m_entities.pop_back();
```

Comment:
If you use unique pointer, when you call pop_back, the delete will get called automatically by the destructor of unique pointer